rfcs/0042-onednn-integration-via-byoc.md

```diff
@@ -0,0 +1,115 @@
+
+Currently, the BYOC homepage provides a simple example of integrating DNNL(naming to oneDNN nowadays) into TVM, but the performance is far away from both TVM autoscheduler and MXNet-oneDNN due to the following main reasons:
+- Non-optimal layout was used in dnnl ops. 
+- Insufficient subgraph partitioning.
+- Unnecessary overhead due to memory copy from tensor to dnnl memory buffer or vice versa.
+
+# Guide-level explanation
+
+We have already solved the above issues and observed the performance benefits by comparing with either MXNet-oneDNN or TVM-autoscheduler on several popular workloads like ResNet50_v1b,  InceptionV3, VGG11_bn in several scenarios including latency (Figure 1, single instance with 28 cores and bs=1), throughput (Figure 2, single instance with 28 core and bs=32) and real-time (Figure 3, 7 instances with 4core per each and bs=1) mode.
+
+## *Note
+[Note]: ##Note
+
+Hardware config
+- Intel(R) Xeon(R) Platinum 8280L CPU @ 2.70GHz
+
+Compilation config
+- g++ 7
+- 'llvm -mcpu=cascadelake -model=platinum-8280'
+- TVM commitID: 19b23b9
+- MXNet version: V1.8.0
+- OneDNN version: V1.7 / V2.4
+
+Runtime config
+- 20 warm-up and 100 batches
+
+![Figure 1 latency scenario](https://github.com/crazydemo/tvm-rfcs/blob/main/rfcs/assets/0042/latency.png)
+
+![Figure 2 Throughput scenario](https://github.com/crazydemo/tvm-rfcs/blob/main/rfcs/assets/0042/throughput.png) 
+
+![Figure 3 Real-time scenario](https://github.com/crazydemo/tvm-rfcs/blob/main/rfcs/assets/0042/real-time.png)
+
+# Reference-level explanation
+This proposal aims to provide a new approach to integrate oneDNN into TVM via DNNL JSON codegen/runtime by applying the following adjustments to tackle the aforementioned issues: 
+- Register a new “alter_op_layout” function for dnnl to get the optimal layouts for dnnl ops with a new layout auto-query function in Relay.
+- Add a custom pass to rewrite “Conv-Add-Add- ReLu” pattern into “Conv-Add- ReLu” to better handle the pattern comes from BatchNorm Folding (“Conv-bias_add-BN-ReLu”).
+- Add a new pattern “Conv-Add-Sum-ReLu” for the fusion.
+- Remove the unnecessary memory copy in “dnnl_json_runtime.cc” with pointer assignment only.
+
+We have enhanced and updated the support. Currently, the following ops/post-op fusion/datatype are enhanced/added, as well as some CV models are verified with the new oneDNN backend, we’re going to cover more ops/datatypes and models (denoted with *) in the next step. 
+
+## Ops
+- nn.conv2d
+- nn.dense
+- nn.relu
+- nn.max_pool2d
+- nn.avg_pool2d
+- matrix multiplication *
+- nn.conv1d *
+- nn.conv3d *
+- depthwise conv *
+
+## Post-Op Fusions
+- conv2d_bias_sum_relu
+- conv2d_bias_relu
+- conv2d_bias
+- dense_bias_relu
+- dense_bias
+- Eltwise Post-op *
+- Depthwise *
+- Binary *
+- PReLu *
+
+## Datatype
+- Float32
+- BF16 *
+- INT8 *
+
+## Verified CV Models (from gluoncv)

```

Comment:
Bert is not a CV models. Let's remove `CV`

Reply:
modified.